src/plugins_types.h

```diff
@@ -129,15 +129,16 @@ LY_ERR ly_type_check_hints(uint32_t hints, const char *value, size_t value_len,
  * Use only in implementations of ::ly_type_store_clb which provide all the necessary parameters for this function.
  *
  * @param[in] ctx libyang context.
+ * @param[in] ctx_node Context node to determine the module if needed (unprefixed data in specific @p format)
```

Comment:
I see, this may just be a working solution even for non-prefixed node names. But it needs much more format-specific details because `ctx_node` is confusing and even incorrect, I think.

For schema formats, I would refer to it as *current node* because that would correspond to how it is called in YANG RFC. It is the schema node closest to the statement, whatever it is. And it must always be defined in the same module that the statement is in (so it does not exist for a leafref path in a typedef, for example).

For JSON format, it is simply the resolved parent node in the path. Is not used if there is no parent since then the node must be prefixed. The function could maybe check this, that it is not accessing NULL `ctx_node` and write the correct format-specific error.

And it is not used for XML, but that should probably be mentioned, too.

Reply:
I've tried to find similar functionality as provide ly_type_resolve_prefix(), but despite there are several similar use cases, there are always some important differences. So I've just renamed the function to `ly_type_identity_module()` to show its connection to identity types.